src/osemgrep/cli_scan/Scan_subcommand.ml

```diff
@@ -267,14 +328,106 @@ let run_scan_files (conf : Scan_CLI.conf) (profiler : Profiler.t)
                    You may need to acquire a different binary."
             | Some invoke_semgrep_core_proprietary ->
                 let roots = conf.target_roots in
-                invoke_semgrep_core_proprietary roots conf.engine_type)
+                invoke_semgrep_core_proprietary roots ~diff_config
+                  conf.engine_type)
       in
       invoke_semgrep_core
         ~respect_git_ignore:conf.targeting_conf.respect_git_ignore
-        ~file_match_results_hook conf.core_runner_conf filtered_rules errors
-        targets
+        ~file_match_results_hook conf.core_runner_conf rules errors targets
+    in
+    let exn_and_matches =
+      match conf.targeting_conf.baseline_commit with
+      | None ->
+          Profiler.record profiler ~name:"core_time"
+            (core targets filtered_rules)
+      | Some baseline_commit ->
+          (* diff scan mode *)
```

Comment:
This can be extracted in a separate function as the code of run_scan_files() is already very long.
You can add a section (*************) with Differential scanning as a title and put the 2 functions you wrote in it, with
some short explanations before.

Reply:
Done.